### PR TITLE
Init sig for SP factorization

### PR DESCRIPTION
**Pull request by HaiwangYu**
As discussed yesterday in [WireCell meeting](https://indico.bnl.gov/event/8210/), push this to WireCell organization as a hub for collaborative development.

Note there is a 'Util' in Aux. I can merge its content into the 'TensUtil' in @brettviren 's [dev branch](https://github.com/brettviren/wire-cell-toolkit/tree/issue54) after that one is merged.

The `Eigen::Map` also copies. Not sure why it is about 4 times faster than `memcpy` (1.8ms vs 6.1ms). Maybe Eigen uses AVX or multi-channel memory.
https://stackoverflow.com/questions/2963898/faster-alternative-to-memcpy
I may study this a bit more.

### CODE REVIEW

File: aux/inc/WireCellAux/Util.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+template<typename ElementType>
+inline Eigen::Array<ElementType, Eigen::Dynamic, Eigen::Dynamic> itensor_to_eigen_array(const ITensor::pointer iten)
+{
+    if (!iten)
+    {
+        THROW(ValueError() << errmsg{"(iten == null"});
+    }
+
+    if (iten->shape().size() != 2)
```
Comment (brettviren) [inline aux/inc/WireCellAux/Util.h:31]: ITensors can be arbitrary shape and Eigen matrices and arrays can be 1D or 2D and Eigen tensors can be ND (these are in Eigen-unsupported).

I wonder if this converter can be made to handle at least 1D or 2D

If the exact signature can't handle this diversity we can make different template specializations.

This comment also applies to the inverse function below.

File: aux/test/test_eigen.cxx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    typedef float TestType;
+
+    // Eigen::Array<TestType, Eigen::Dynamic, Eigen::Dynamic> iarr =
+    // Eigen::Array<TestType, Eigen::Dynamic, Eigen::Dynamic>::Random(2, 3);
+    Eigen::Array<TestType, 2, 3> iarr;
+    iarr << 1, 2, 3,
+            4, 5, 6;
+    std::cout << iarr << "\n";
+
+    auto iten = Aux::eigen_array_to_itensor<TestType>(iarr);
```
Comment (brettviren) [inline aux/test/test_eigen.cxx:44]: Besides `cout` lines for a one time "by eye" test, it's good to litter the code with asserts.  `WireCellUtil/Testing.h` provides `Assert()` and `AssertMsg()` to provide some value-add beyond standard `assert()`.

File: sig/src/Decon2DResponse.cxx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+void Sig::Decon2DResponse::configure(const WireCell::Configuration &cfg)
+{
+    m_cfg = cfg;
+}
+
+namespace
+{
+std::string dump(const ITensorSet::pointer &itens) {
```
Comment (brettviren) [inline sig/src/Decon2DResponse.cxx:33]: Maybe can go to `Util.h`?

**Comment by HaiwangYu** [comment]
Thanks @brettviren  again for the suggestions. I added a content dump function for `ITensor` trying to mimic numpy style.

Output from the `test_dump.cxx`:
image

File: aux/inc/WireCellAux/Util.h (hunk 1 above)
Comment (HaiwangYu) [reply aux/inc/WireCellAux/Util.h:98]: Will try to improve this in future versions.

File: aux/test/test_eigen.cxx (hunk 2 above)
Comment (HaiwangYu) [reply aux/test/test_eigen.cxx:21]: There was one, forgot to uncomment before pushing. Fixed.

File: sig/src/Decon2DResponse.cxx (hunk 3 above)
Comment (HaiwangYu) [reply sig/src/Decon2DResponse.cxx:33]: Generalized for n-dim and put in `Util`.

**Pull request merged**